### PR TITLE
Add storybook for floating widget options

### PR DESCRIPTION
**Pull request by GerardasB**
## Changes

This PR adds a story to showcase different [CanFloatWidgetOptions ](https://www.itwinjs.org/reference/appui-react/widget/canfloatwidgetoptions).
This PR also disables user selection when dragging a widget tab to avoid unintentional selection.
Documentation of related APIs is updated.

## Testing

N/A

### CODE REVIEW

**Comment by joehenry9498** [comment]
All files look good and the new storybook is working as expected

**Review by joehenry9498 — APPROVED** [review]

File: docs/storybook/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "type": "module",
   "scripts": {
     "build:storybook": "npm run copy:webfont && cross-env NODE_OPTIONS=\"--max-old-space-size=8192\" storybook build",
-    "dev": "npm run copy:webfont && storybook dev -p 6006",
+    "start": "npm run copy:webfont && storybook dev -p 6006",
```
Comment (raplemie) [inline docs/storybook/package.json:8]: thanks :)

**Review by raplemie — APPROVED** [review]
I got an issue where the "release" of a tab would not work (after releasing the button when unmerging a tab, the tab would still follow the cursor) It did that like 5 times in different occasions, I doubt this has anything to do with user-select though)

File: docs/storybook/src/components/ToolbarComposer.stories.tsx [hunk 2]
```diff
@@ -11,7 +12,7 @@ import {
   ToolbarUsage,
   UiFramework,
 } from "@itwin/appui-react";
-import { ToolbarComposer } from "@itwin/appui-react/src/appui-react/toolbar/ToolbarComposer";
+import { ConditionalIconItem, IconHelper } from "@itwin/core-react";
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply docs/storybook/src/components/ToolbarComposer.stories.tsx:15]: Linter is enabled for stories, there is no rule for imports though.

File: docs/storybook/src/widget/CanFloatOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  return {
+    id: "widgets",
+    provideWidgets: () => {
+      const floatingWidget = {
+        id: "w1",
+        content: <>Widget 1 content </>,
+        defaultState: WidgetState.Floating,
+        canFloat: props,
+      };
+      if (props.containerId) {
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply docs/storybook/src/widget/CanFloatOptions.tsx:30]: Changed to add both widgets in all stories.

**Comment by GerardasB** [comment]
> I got an issue where the "release" of a tab would not work (after releasing the button when unmerging a tab, the tab would still follow the cursor) It did that like 5 times in different occasions, I doubt this has anything to do with user-select though)

To reproduce this consistently it seems you need to select some HTML element (i.e. _general.no-content_) and then start dragging the widget by its title bar.

**Pull request merged**